server/portal/apps/datafiles/views.py

```diff
@@ -38,6 +38,12 @@ def get(self, request):
         default_system = user_systems[settings.PORTAL_DATA_DEPOT_LOCAL_STORAGE_SYSTEM_DEFAULT]
         response['default_host'] = default_system['host']
 
+        for system in response['system_list']:
+            try:
+                system['definition'] = request.user.agave_oauth.client.systems.get(systemId=system['system'])
```

Comment:
should we create a ticket to optimize this?   right now, we have 2 tapis systems right (frontera home and longhorn home) but in the future there could be many.  so maybe better to make a lot of calls to get the system definitions at the same time (like with `concurrent.futures.ThreadPoolExecutor`)?

Reply:
Maybe? Not sure we want to get into that at this juncture though. I think it's better to wait for tapis v3 and see if they make a speed optimization to these calls.

Reply:
Could we separate the definition request out from the system list request? I worry the systemList request will begin taking too much time

Reply:
Can we make this an optimization for later?

Reply:
Reasoning - there are just many things that our workbench needs or could use before it loads that we are putting into a bunch of different sagas at this point, and possibly loading and re-loading depending on the view that needs it. I think we need a general solution for something like this, possibly related to the LoadingSpinner for the workbench state. Like, a saga that runs upon workbench load, triggering all the things we need to load such as system definitions (however we want to do it) and then we can let everything downstream just share the state once the workbench has loaded.

Reply:
for some systems lists, we'll have 3 or 4 tapis system gets (2 homes, community, public and project). so we would have to wait for ~4 calls to sequential calls to tapis.  

if we do leave it here, optimizing with something like in concurrent.futures like ThreadPoolExecutor would at least allow us to do those 3 or 4 tapis-system-gets concurrently .  

That said, the speed of this GET (`api/datafiles/systems/list/`) seems to be crucial as we can't list our storage systems in Data Files till its completed.  and an extra delay for an optional feature for single portal, then separating the system(s) definition requests from here makes sense as well.

[This is not any guidance on what we should do but just putting some thoughts down to continue the conversation :) ]